### PR TITLE
[GAL-340] [GAL-341] Tezos Login flow

### PR DESCRIPTION
**Pull request by jakzaizzat**
## Changes

- Update coming soon styling
- Added tezos wallet for login flow
- Enabled tezos button with shortcut - `CTRL` + `t`
- Show invalid address on error message

## Screenshot / Demo

**Coming soon button**

CleanShot 2022-08-26 at 09 25 35

**Sign up with address not in allowlist**

https://user-images.githubusercontent.com/4480258/186796958-76fa11a8-aae7-4f4b-9b27-7b2e408a432a.mp4

**Sign up with address in the allowlist**

https://user-images.githubusercontent.com/4480258/186797113-7555b4b1-ee55-479b-939d-97e8700f41a3.mp4

**Sign up with existing user**

https://user-images.githubusercontent.com/4480258/186797238-2508249d-1537-4d23-9c32-45045c29bf20.mp4

**Show address in the error message**

CleanShot 2022-08-29 at 17 01 38

CleanShot 2022-08-29 at 17 04 46

### CODE REVIEW

**Comment by kaitoo1** [comment]
lets go!!!! I can't believe my eyes

**Comment by Robinnnnn** [comment]
image

if the address is not on the allowlist, could we specify the address to the user?

for example, below is the error for an ETH address that's not on the allowlist:

image

File: src/components/WalletSelector/multichain/MultichainWalletSelector.tsx [hunk 1]
```diff
@@ -72,6 +85,23 @@ export function MultichainWalletSelector({ connectionMode = AUTH, queryRef }: Pr
...
   }
 
+  if (selectedAuthMethod === supportedAuthMethods.tezos) {
+    // if (connectionMode === ADD_WALLET_TO_USER) {
+    //   return (
+    //     <StyledWalletSelector>
+    //       <EthereumAddWallet queryRef={query} reset={reset} />
+    //     </StyledWalletSelector>
+    //   );
+    // }
```
Comment (Robinnnnn) [inline src/components/WalletSelector/multichain/MultichainWalletSelector.tsx:95]: this will be a separate PR?

Comment (Robinnnnn) [inline src/components/WalletSelector/multichain/MultichainWalletSelector.tsx:95]: this will be a separate PR?

File: src/components/WalletSelector/multichain/MultichainWalletSelector.tsx [hunk 2]
```diff
@@ -100,7 +130,15 @@ export function MultichainWalletSelector({ connectionMode = AUTH, queryRef }: Pr
...
+      <WalletButton
+        label="Tezos"
+        icon="tezos"
+        disabled={!isTezosConnectEnabled}
+        onClick={() => {
+          console.log('connecting to tezos via beacon');
+          setSelectedAuthMethod(supportedAuthMethods.tezos);
+        }}
+      />
       {/* <WalletButton label="Solana" icon="solana" disabled /> */}
```
Comment (Robinnnnn) [inline src/components/WalletSelector/multichain/MultichainWalletSelector.tsx:142]: let's show the solana option too! with "COMING SOON" 😎

File: src/components/WalletSelector/multichain/supportedAuthMethods.ts [hunk 3]
```diff
@@ -1,17 +1,21 @@
-export type SupportedAuthMethodKey = 'ethereum' | 'gnosisSafe';
+export type SupportedAuthMethodKey = 'ethereum' | 'gnosisSafe' | 'tezos';
```
Comment (Robinnnnn) [inline src/components/WalletSelector/multichain/supportedAuthMethods.ts:1]: nit: could we consolidate usage of these consts with the one below?

image

File: src/components/WalletSelector/multichain/tezos/TezosAuthenticateWallet.tsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+import { DAppClient, RequestSignPayloadInput, SigningType } from '@airgap/beacon-sdk';
+import { char2Bytes } from '@taquito/utils';
+
+type Props = {
+  reset: () => void;
+};
+
+export const TezosAuthenticateWallet = ({ reset }: Props) => {
+  const beaconClient = useMemo(() => {
+    return new DAppClient({ name: 'Gallery' });
```
Comment (Robinnnnn) [inline src/components/WalletSelector/multichain/tezos/TezosAuthenticateWallet.tsx:26]: i believe this might initiate a new dapp client each time the component unmounts / remounts. i think it would be safest to simply define the client above the component:

```js
const beaconClient = new DAppClient({ name: 'Gallery' });

export const TezosAuthenticateWallet ...
```

File: src/components/WalletSelector/multichain/tezos/TezosAuthenticateWallet.tsx [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+        setError(normalizeError(error));
+      }
+    }
+
+    void authenticate();
+  }, [attemptAuthentication, trackSignInError, beaconClient]);
+
+  if (error) {
+    return (
+      <EthereumError
```
Comment (Robinnnnn) [inline src/components/WalletSelector/multichain/tezos/TezosAuthenticateWallet.tsx:130]: nit: this is called `EthereumError`, maybe we could call it `WalletError` or something more generic

**Review by Robinnnnn — COMMENTED** [review]
just left some small comments! SOOOOO EXCITED!!!

File: src/components/WalletSelector/AddWalletPending/AddWalletPendingDefault.tsx [hunk 6]
```diff
@@ -118,7 +118,7 @@ function AddWalletPendingDefault({
         setPendingState(PROMPT_SIGNATURE);
 
         trackAddWalletAttempt(userFriendlyWalletName);
-        const { nonce, user_exists: userExists } = await createNonce(address);
+        const { nonce, user_exists: userExists } = await createNonce(address, 'Ethereum');
```
Comment (kaitoo1) [inline src/components/WalletSelector/AddWalletPending/AddWalletPendingDefault.tsx:121]: nit: maybe 'Ethereum' can be a const since it's used in a lot of places?

File: src/components/WalletSelector/multichain/tezos/TezosAuthenticateWallet.tsx [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+  const trackSignInError = useTrackSignInError();
+
+  /**
+   * Auth Pipeline:
+   * 1. Fetch nonce from server with provided wallet address
+   * 2. Sign nonce with wallet (metamask / walletconnect / etc.)
+   * 3a. If wallet exists, log user in
+   * 3b. If wallet is new, sign user up
+   */
+  const attemptAuthentication = useCallback(
```
Comment (kaitoo1) [inline src/components/WalletSelector/multichain/tezos/TezosAuthenticateWallet.tsx:51]: 👏👏👏

**Review by kaitoo1 — APPROVED** [review]

**Review by Robinnnnn — APPROVED** [review]
